jlh/WeatherApp/app/src/main/java/com/e_conomic/weatherapp/domain/mappers/ForecastDataMapper.kt

```diff
@@ -0,0 +1,33 @@
+package com.e_conomic.weatherapp.domain.mappers
+
+import com.e_conomic.weatherapp.data.Forecast
+import com.e_conomic.weatherapp.data.ForecastResult
+import com.e_conomic.weatherapp.domain.model.ForecastList
+import java.text.DateFormat
+import java.util.*
+import com.e_conomic.weatherapp.domain.model.Forecast as ModelForecast
+
+class ForecastDataMapper {
+
+    fun convertFromDataModel(forecast: ForecastResult): ForecastList {
+        return ForecastList(forecast.city.name, forecast.city.country,
+                convertForecastListToDomain(forecast.list))
+    }
+
+    private fun convertForecastListToDomain(forecastList: List<Forecast>): List<ModelForecast> {
+        return forecastList.map { convertForecastItemToDomain(it) }
+    }
+
+    private fun convertForecastItemToDomain(forecast: Forecast): ModelForecast {
+        return ModelForecast(convertDate(forecast.dt), forecast.weather[0].description,
```

Comment:
I guess it could be empty. What would you recommend to do here in that case? Log it, put a default value or something?